### ISSUE TITLE
Making R-Instat look modern?

### ISSUE THREAD

**Issue by shadrackkibet** [africanmathsinitiative/R-Instat#3898]
I was wondering if this is easy to implement perhaps someone has suggestions how we can do this to make R-Instat look modern? In Spss we have icons on the side of menu items .Can we have something similar to this in R-Instat?

![image](https://user-images.githubusercontent.com/26301973/30170393-b718d2f2-93f7-11e7-829f-d25f86b2a522.png)

**Comment by dannyparsons** [comment]
I think that this is a built in feature we could use. I'm only guessing because there is a space for an icon on the menu items.
Personally, I'm not sure how much this enhances the look of the software but maybe it would be welcomed by our audience.
What do others think?

**Comment by dannyparsons** [comment]
I was also wondering about ribbons like in the new Office programs. I'm not sure if this is suited for our menu system or if it's even easy to implement.

**Comment by dannyparsons** [comment]
What I do like from the SPSS image is the large icon buttons. Since we don't have that many I think they could be much bigger and then clearer.

**Comment by shadrackkibet** [comment]
Having such icons at the side of our items on **Describe>specific>scatterplot,lineplot,Histogram**  just to mention few places .Personally i feel this is one thing that will enhance the look of R-instat.

**Comment by rdstern** [comment]
I like this discussion.
I would like an improved toolbar, with (at least the option of) larger icons and also a few more icons.  Ones I would use include switches between the different views, e.g. with the column metadata, log file, etc.
Might some icons also produce a (small?) menu?  I think that is what happens in the new jamovi software.  Have a look here https://www.jamovi.org/user-manual.html 

The jamovi manual shows screen shots of jamovi and you can click in some of the screen to see the software "in action".  I assume it would be quite easy for us to embed some of our short videos in this way.

When Stats4SD saw R-Instat and jamovi alongside they said that jamovi was modern - and R-Instat wasn't.  I think other neat elements in jamovi included what could be done really slickly on output through clicking in the output screen, Ffor us that might mean you could click on a ggplot and go straight to the interactive version..  jamovi does look neat and clean (though there is very little in the actual software to date).  It may show that good marketing is really important, i.e. the image in this case.  However I still feel that once sold the ever-growing substance in R-Instat is also really important.

**Comment by lilyclements** [comment]
I wouldn't use SPSS as a "modern" guideline per-se, however, I agree that the likes of Jamovi is a good "modern" template (which just uses a template similar to that of Microsoft!). Therefore, I'm not sure on icons next to a menu item (it's very easy to do - right click and press "Set Image..") because, to me, our current menu layout is a little outdated and so we need to do more than add icons to the left hand side!

I think the idea of having ribbons with nice big icons with them will really help the software look more up-to-date - as well as a lot "sleeker" and easier to navigate
I agree about the larger icons too for the new file/save/copy/paste/etc options too for the same reasons!

**Comment by dannyparsons** [comment]
Having thought about it, I don't think I should have brought up the menu vs ribbons discussion in this thread. I think it's good to consider this but I think it goes beyond making R-Instat more modern since it will really change how users interact with the software and would need quite a lot of thought before deciding to implement.
I think we would agree to separating out making R-Instat look more modern/friendly is different from changing functionality to match "modern" software. So I will move the menu/ribbon discussion to here #3903 and discussion on other ideas can continue.

**Comment by dannyparsons** [comment]
I agree with the larger toolbar icons, but also we want to remember users on small screens. A big complaint of Office is that the menus and toolbars become too distracting and take too much space from the main area on smaller screens.

**Comment by rdstern** [comment]
I have put the end of the procurement project as a milestone for something to be done here.  I assume we could at least have a few extra toolbar icons.
I also don't like the current situation where delete the output window is just next to the use last dialogue.  
Just a few new icons (with tool-tips would add something.  (This might also relieve us from making the right-click menu too long.)
Important ones for me include the 3 different windows, namely column metadata, data frame metadata and log window.  (I haven't used the script window).  Then I would also possibly include the Windows Reset to Default setting.
Might there also be a "Maximise current window"?
And Show/Hide R-Commands in Output Window
I wonder about also including Tools > Options > Data.  (Would it be easy for that facility to have the Option for just the current data frame or all data frames.  Currently it is only for all data frames.

I looked for toolbar icons on Google and there seem to be a lot of possibilities.  I could find opions below for most of the ideas above.  More important maybe to make sure they resize sensibly, so they can be used on small screens.  If necessary we stick with the current size.
![image](https://user-images.githubusercontent.com/11226469/30772166-eedd0390-a04d-11e7-8c76-2475318ebd89.png)

**Comment by shadrackkibet** [comment]
@rdstern I suggest the addition of icons on the side of menu items in R-Instat. This is something that is done in other statistical software for example **SPSS**. See below examples

![image](https://user-images.githubusercontent.com/26301973/110333773-104b7480-8033-11eb-8188-7d548dd3eaa8.png)
![image](https://user-images.githubusercontent.com/26301973/110333848-235e4480-8033-11eb-9775-9fd5854dd9a7.png)
![image](https://user-images.githubusercontent.com/26301973/110333888-32dd8d80-8033-11eb-8407-9bc75297657d.png)

**Comment by rdstern** [comment]
@shadrackkibet I like that idea.  What does spss do on more specialised dialogues?  I quite like the idea also as we start translations.

**Comment by shadrackkibet** [comment]
They have icons for almost all dialogs.

**Comment by shadrackkibet** [comment]
Icons can be added to the dialogs as well. For example

![image](https://user-images.githubusercontent.com/26301973/110341849-d763cd80-803b-11eb-91cd-00803ec45aba.png)

**Comment by rdstern** [comment]
We did some work earlier when updating the toolbar.  Rachel - who worked on it says of possible icons:

"There are websites with thousands of them. You can get some for free and others you pay very small fees for the copyright. "

So, I am quite tempted to try making some progress on this?
